kml/serdes_gpx.cpp

```diff
@@ -492,15 +492,31 @@ void SaveTrackData(Writer & writer, TrackData const & trackData)
     writer << "</color>\n" << kIndent2 << "</extensions>\n";
   }
   bool const trackHasAltitude = TrackHasAltitudes(trackData);
-  for (auto const & line : trackData.m_geometry.m_lines)
+  auto const & geom = trackData.m_geometry;
+  for (size_t lineIndex = 0; lineIndex < geom.m_lines.size(); ++lineIndex)
   {
+    auto const & line = geom.m_lines[lineIndex];
+    auto const & timestampsForLine = geom.m_timestamps[lineIndex];
+    auto const lineHasTimestamps = geom.HasTimestampsFor(lineIndex);
+
     writer << kIndent2 << "<trkseg>\n";
-    for (auto const & point : line)
+
+    for (size_t pointIndex = 0; pointIndex < line.size(); ++pointIndex)
     {
+      auto const & point = line[pointIndex];
       auto const [lat, lon] = mercator::ToLatLon(point);
+      
       writer << kIndent4 << "<trkpt lat=\"" << CoordToString(lat) << "\" lon=\"" << CoordToString(lon) << "\">\n";
+      
       if (trackHasAltitude)
         writer << kIndent6 << "<ele>" << CoordToString(point.GetAltitude()) << "</ele>\n";
+
+      if (lineHasTimestamps)
+      {
+        CHECK_EQUAL(line.size(), timestampsForLine.size(), ());
```

Comment:
This check should be after L500, not here in the loop:
```
if (lineHasTimestamps)
  CHECK_EQUAL(line.size(), timestampsForLine.size(), ());
```

Reply:
Fixed! Thanks